watch_cron_logs.pl

```diff
@@ -60,18 +60,26 @@ ($)
     }
 }
 
-# Slide every daily directory up by one and delete 8th day if there
-for $i (reverse (0 .. $n_days-1)) {
-    $i1 = sprintf "%d", $i+1;
-    run "mv $logs/daily.$i $logs/daily.$i1" if -e "$logs/daily.$i";
+my $last_master_file = "$logs/daily.0/$master_log";
+my $same_day_update = 0;
+if (-e $last_master_file){
+    my $creation_time = (stat $last_master_file)[10];
+    my $now = time;
+    if (($now - $creation_time) < 86400){
```

Comment:
Using a time difference here of a day does run the risk of an under-increment, where a truly daily cron job ends up with an appended log instead of incrementing the daily directories.  I think one could safely reduce the 86400 by a bit, or one could really check the calendar day of "now" and the calendar day of the creation of the file instead of a difference value.